Gems/Terrain/Code/Source/TerrainRenderer/ClipmapBounds.h

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/base.h>
+#include <AzCore/Math/Aabb.h>
+#include <AzCore/Math/Vector2.h>
+#include <AzCore/std/containers/vector.h>
+#include <TerrainRenderer/Vector2i.h>
+#include <TerrainRenderer/Aabb2i.h>
+
+namespace Terrain
+{
+
+    struct ClipmapBoundsDescritor
+    {
+        //! Size of the clipmap in width / height.
+        uint32_t m_size = 1024;
+
+        //! Current center location of the clipmap
+        AZ::Vector2 m_center = AZ::Vector2::CreateZero();
+
+        //! Unused area around edges of the clipmap. Center point must be moved by
+        //! at least this much before an update is triggered. This allows for larger
+        //! but less frequent updates.
+        uint32_t m_margin = 4;
+
+        //! Scale of the clip map compared to the world. A scale of 0.5 means that
+        //! a clipmap of size 1024 would cover 512 meters.
+        float m_scale = 1.0f;
```

Comment:
I don't want to encode a specific length unit into this class because it doesn't really care.

One idea to make this clearer would be to have two sizes - a clipmap size and a world size. What do you think about that?

Reply:
I think the class _does_ care about the unit though. It has APIs like "GetWorldBounds", and this scale is defined to be the scale to convert to the world. So while it might not specifically care that the unit is meters, it _does_ care that the unit matches the world unit, which is meters.

IMO having it as a scale is clearer than having two sizes, it's just "m_scale" to me isn't clear if the scale represents "world to clip" or "clip to world". Even in writing my comment, I had to think it through to make sure "metersPerPixel" was correct vs "pixelsPerMeter". 

Guess as a compromise, how about "m_clipToWorldScale" ?

Reply:
m_clipToWorldScale works for me. I agree that this should be whatever unit the world is in, just that it doesn't have to be used for a world that happens to be in meters.